### ISSUE TITLE
Upgrading to latest pyright causes new error - false positive? (1.1.251 from 1.1.246)

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3534]
### Discussed in https://github.com/microsoft/pyright/discussions/3530

<div type='discussions-op-text'>

<sup>Originally posted by **jab** June  1, 2022</sup>
As part of my monthly maintenance work, I just upgraded from pyright 1.1.246 to 1.1.251, the current latest stable release. The new version emits a new error at a couple call sites, unless I include the `__NO_KW: Mapping[str, Any]` workaround you can see in the first line of [the "upgrade pyright" commit](https://github.com/jab/bidict/commit/f31a6f88f3c9899808286767d97529f5175cad7c) (this is only a 5-line diff and it's very simple, so don't be scared to click through:).

Here are examples of the new error that is emitted without the workaround:

```console
  /Users/jab/src/bidict/bidict/_bidict.py:93:9 - error: Argument of type "dict[str, VT@BidictBase]" cannot be assigned to parameter "kw" of type "Mapping[str, VT@MutableBidict]" in function "_update"
    TypeVar "_VT_co@Mapping" is covariant
      Type "VT@BidictBase" cannot be assigned to type "VT@MutableBidict" (reportGeneralTypeIssues)
  /Users/jab/src/bidict/bidict/_bidict.py:179:13 - error: Argument of type "dict[str, VT@BidictBase]" cannot be assigned to parameter "kw" of type "Mapping[str, VT@MutableBidict]" in function "_update"
    TypeVar "_VT_co@Mapping" is covariant
      Type "VT@BidictBase" cannot be assigned to type "VT@MutableBidict" (reportGeneralTypeIssues)
```

Since pyright didn't emit any error before, and since the latest stable release of mypy also is not emitting any error without the workaround, I'm wondering if this is a regression in pyright?

If not, then:
1. Is there a better solution than the workaround I came up with?
2. Is there a bug in the mypy tracker for its associated false negative?

Thanks,
Josh</div>

**Comment by erictraut** [comment]
Here is a minimal repro:
```python
from typing import Generic, Iterable, TypeVar

T = TypeVar('T')

class IterableProxy(Iterable[T]): ...

class Parent(Generic[T]):
    def m1(self, v: Iterable[T] = IterableProxy()) -> None:
        ...

class Child(Parent[T]):
    def m2(self) -> None:
        self.m1()
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.252, which I just published. It will also be included in the next release of pylance.
